### PR TITLE
Fix JSONField deprecation warnings

### PR DESCRIPTION
**Pull request by daviddavis**
fixes #9112

Please be sure you have read our documentation on creating PRs:
https://docs.pulpproject.org/contributing/pull-request-walkthrough.html

### CODE REVIEW

File: pulpcore/app/migrations/0023_change_exporter_models.py [hunk 1]
```diff
@@ -20,7 +19,7 @@ class Migration(migrations.Migration):
                 ('pulp_id', models.UUIDField(default=uuid.uuid4, editable=False, primary_key=True, serialize=False)),
                 ('pulp_created', models.DateTimeField(auto_now_add=True)),
                 ('pulp_last_updated', models.DateTimeField(auto_now=True, null=True)),
-                ('params', django.contrib.postgres.fields.jsonb.JSONField(null=True)),
+                ('params', models.JSONField(null=True)),
```
Comment (fao89) [inline pulpcore/app/migrations/0023_change_exporter_models.py:22]: I think you shouldn't change the migrations, only the models, and then migrate

File: pulpcore/app/models/repository.py [hunk 2]
```diff
@@ -262,6 +260,7 @@ class Remote(MasterModel):
         connect_timeout (models.FloatField): Value for aiohttp.ClientTimeout.connect
         sock_connect_timeout (models.FloatField): Value for aiohttp.ClientTimeout.sock_connect
         sock_read_timeout (models.FloatField): Value for aiohttp.ClientTimeout.sock_read
+        headeres (models.JSONField): Headers set on the aiohttp.ClientSession
```
Comment (dralley) [inline pulpcore/app/models/repository.py:263]: Typo

File: pulpcore/app/migrations/0023_change_exporter_models.py (hunk 1 above)
Comment (mdellweg) [reply pulpcore/app/migrations/0023_change_exporter_models.py]: If you don't change the migrations, does that need we can never drop the django.contrib dependency? Is that even a thing?

Comment (dralley) [reply pulpcore/app/migrations/0023_change_exporter_models.py]: Change the models and then do whatever makemigrations does ^

Comment (daviddavis) [reply pulpcore/app/migrations/0023_change_exporter_models.py]: I'm not sure but it doesn't trigger any deprecation warnings so I think we're good.

Comment (fao89) [reply pulpcore/app/migrations/0023_change_exporter_models.py]: I'm seeing some jsonb issues on pulp_ansible, I'm investigating
https://github.com/pulp/pulp_ansible/runs/3125697259?check_suite_focus=true#step:11:1681

**Review by dralley — APPROVED** [review]

File: pulpcore/app/migrations/0023_change_exporter_models.py (hunk 1 above)
Comment (fao89) [reply pulpcore/app/migrations/0023_change_exporter_models.py]: failure is not related to this

**Review by mdellweg — APPROVED** [review]
Looks good to me. And keeping the migrations intact seems to be correct according to this quote from django:
'django.contrib.postgres.fields.JSONField is removed except for support in historical migrations.'

**Pull request merged**